### PR TITLE
Donation state machine and notifications

### PR DESCRIPTION
**Pull request by devton**
- Added donation state machine to better history of transaction status changes.
- Added notifications support to Donation (waiting_payment_donation, paid_donation, refused_donation)
- Added unique index to donations.transaction_id

### CODE REVIEW

File: app/models/donation.rb [hunk 1]
```diff
@@ -181,4 +197,38 @@ def subscribe_attributes
     return_attributes[:CITY] = self.activist.city if self.activist and self.activist.city
     return_attributes
   end
+
+  def notify_activist(template_name, template_vars = {}, auto_deliver = true)
+    Notification.notify!(
+      activist_id,
+      template_name,
+      default_template_vars.merge(template_vars),
+      community_id,
```
Comment (luizalbsilva) [inline app/models/donation.rb]: where does this community_id comes from? Maybe from widget.community.id?

File: db/migrate/20170525191310_add_unique_index_to_donations_transaction_id.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUniqueIndexToDonationsTransactionId < ActiveRecord::Migration
+  def change
+    add_index :donations, :transaction_id, unique: true
```
Comment (luizalbsilva) [inline db/migrate/20170525191310_add_unique_index_to_donations_transaction_id.rb]: You are going to get a little trouble if you try to run this on production... You'll have to fix a record (in production). 

Take a look in this 2 donations: 3903 and 3902

Comment (devton) [inline db/migrate/20170525191310_add_unique_index_to_donations_transaction_id.rb]: thanks! fixes already

File: app/models/donation.rb (hunk 1 above)
Comment (devton) [inline app/models/donation.rb]: you are right we does not have the community_id just need to use comunity.id, send a commit about this already too

**Comment by devton** [comment]
@luizalbsilva about you suggestion, sometimes we need to change the template copy, so this rake task already do de update on template, and we can rerun every time.
If we put this on migration, we need to create an migration for every text changes on notification.

**Pull request merged**